models/articles.model.js

```diff
@@ -15,7 +15,31 @@ exports.readArticleById = (article_id) => {
     return rows[0];
   });
 };
-exports.readAllArticles = () => {
+exports.readAllArticles = (sort_by = "created_at", order = "DESC") => {
+  const validSorts = [
+    "created_at",
+    "title",
+    "topic",
+    "author",
+    "body",
+    "article_img_url",
+  ];
+  const validOrders = ["DESC", "ASC"];
+  const upperCaseOrder = order.toUpperCase();
+
+  if (!validSorts.includes(sort_by)) {
+    return Promise.reject({
+      status: 400,
+      msg: "Bad request",
+    });
+  }
+  if (!validOrders.includes(upperCaseOrder)) {
+    return Promise.reject({
+      status: 400,
+      msg: "Bad request",
+    });
```

Comment:
Nice approach with the green list! 🥇 Another approach (which only works because `order` can only be 2 values), we could just check if `order === "asc"`:
- if true `order === "ASC"`
- if false `order === "DESC"`
Both work well to prevent bad queries getting into our database 👍